README.md
# vuepress-theme-kb
 


CHANGELOG.md
## 1.3.10 (2019-11-16)



## 1.3.9 (2019-11-16)



## 1.3.8 (2019-11-16)



## 1.3.7 (2019-11-16)



## 1.3.6 (2019-11-16)



## 1.3.5 (2019-11-16)



## 1.3.2 (2019-11-16)



## 1.3.2 (2019-11-10)



## 1.3.2 (2019-11-10)



## [1.3.2](https://github.com/ulivz/vuepress-theme-blog/compare/v1.3.1...v1.3.2) (2019-10-20)


### Features

* support passing comment props via frontmatter ([815faf4](https://github.com/ulivz/vuepress-theme-blog/commit/815faf477cd21ada70da5c7e7aedaabe56d06879))



## [1.3.1](https://github.com/ulivz/vuepress-theme-blog/compare/v1.3.0...v1.3.1) (2019-10-16)


### Bug Fixes

* fix broken content layout ([553dcf4](https://github.com/ulivz/vuepress-theme-blog/commit/553dcf48622e6ea87e97f2ad38144634183067c2))



# [1.3.0](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.6...v1.3.0) (2019-10-15)


### Features

* support sitemap and commenting ([9690689](https://github.com/ulivz/vuepress-theme-blog/commit/9690689595e4e635492ad96c393781b661d3a7fb))



## [1.2.6](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.5...v1.2.6) (2019-10-11)


### Bug Fixes

* solve invalid date on Safari ([b29c1b6](https://github.com/ulivz/vuepress-theme-blog/commit/b29c1b62a89b1a79ac2e1bfafb677496a5c7ea66))
* summary default should be true ([46e503c](https://github.com/ulivz/vuepress-theme-blog/commit/46e503cd7d200e7982bc6d4e98c20feb69759cb2))


## [1.2.5](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.4...v1.2.5) (2019-09-10)


### Bug Fixes

* remove excess space in mobile side ([c4cc56e](https://github.com/ulivz/vuepress-theme-blog/commit/c4cc56e))


### Features

* bump blog plugin to v1.2.5 ([96b16b9](https://github.com/ulivz/vuepress-theme-blog/commit/96b16b9))
* close mobile menu after clicking on content ([78453bc](https://github.com/ulivz/vuepress-theme-blog/commit/78453bc))
* close mobile menu after navigation ([266b0d5](https://github.com/ulivz/vuepress-theme-blog/commit/266b0d5))



## [1.2.4](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.3...v1.2.4) (2019-09-09)


### Bug Fixes

* base url in header ([#9](https://github.com/ulivz/vuepress-theme-blog/issues/9)) ([0d70cd4](https://github.com/ulivz/vuepress-theme-blog/commit/0d70cd4))
* config for summary, summaryLength and pwa (close: [#13](https://github.com/ulivz/vuepress-theme-blog/issues/13))([#14](https://github.com/ulivz/vuepress-theme-blog/issues/14)) ([7d40fe3](https://github.com/ulivz/vuepress-theme-blog/commit/7d40fe3))
* prevent nav from breaking into next line ([#17](https://github.com/ulivz/vuepress-theme-blog/issues/17)) ([377c706](https://github.com/ulivz/vuepress-theme-blog/commit/377c706))



## [1.2.3](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.2...v1.2.3) (2019-07-29)


### Bug Fixes

* <Toc /> component can not display normally (close: [#7](https://github.com/ulivz/vuepress-theme-blog/issues/7))([#11](https://github.com/ulivz/vuepress-theme-blog/issues/11)) ([7d1cc3d](https://github.com/ulivz/vuepress-theme-blog/commit/7d1cc3d))



### Features

* add contact type 'instagram' and 'linkedin' ([#10](https://github.com/ulivz/vuepress-theme-blog/issues/10)) ([ca0590c](https://github.com/ulivz/vuepress-theme-blog/commit/ca0590c))



## [1.2.2](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.1...v1.2.2) (2019-06-23)


### Bug Fixes

* require `modifyBlogPluginOptions` to delete the `archive` classifier ([5a34e26](https://github.com/ulivz/vuepress-theme-blog/commit/5a34e26))
* Vue is undefined ([42118a3](https://github.com/ulivz/vuepress-theme-blog/commit/42118a3))



## [1.2.1](https://github.com/ulivz/vuepress-theme-blog/compare/v1.2.0...v1.2.1) (2019-06-23)


### Bug Fixes

* bump blog plugin to 1.2.1 fix filter bug ([eab7fdd](https://github.com/ulivz/vuepress-theme-blog/commit/eab7fdd))



# [1.2.0](https://github.com/ulivz/vuepress-theme-blog/compare/v1.1.0...v1.2.0) (2019-06-22)


### Features

* paginationComponent option - ability to cutom pagination component ([5518887](https://github.com/ulivz/vuepress-theme-blog/commit/5518887))



# [1.1.0](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.1...v1.1.0) (2019-06-12)



## [1.0.1](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0...v1.0.1) (2019-06-09)


### Features

* pwa option (defaults to false) ([6a5c282](https://github.com/ulivz/vuepress-theme-blog/commit/6a5c282))



# [1.0.0](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0-beta.6...v1.0.0) (2019-06-09)


### Features

* bump deps ([d5468d4](https://github.com/ulivz/vuepress-theme-blog/commit/d5468d4))
* enhance the styls of SW-Update Popup ([#2](https://github.com/ulivz/vuepress-theme-blog/issues/2)) ([25d644f](https://github.com/ulivz/vuepress-theme-blog/commit/25d644f))



# [1.0.0-beta.6](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0-beta.5...v1.0.0-beta.6) (2019-06-06)


### Features

* CSS Fixes and adjustments ([#1](https://github.com/ulivz/vuepress-theme-blog/issues/1)) ([066c1fd](https://github.com/ulivz/vuepress-theme-blog/commit/066c1fd))



# [1.0.0-beta.5](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0-beta.4...v1.0.0-beta.5) (2019-06-05)


### Features

* enhance styles ([19e3841](https://github.com/ulivz/vuepress-theme-blog/commit/19e3841))



# [1.0.0-beta.4](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0-beta.3...v1.0.0-beta.4) (2019-06-05)


### Features

* enhance the styles ([0a2c067](https://github.com/ulivz/vuepress-theme-blog/commit/0a2c067))
* summary & summary option ([7f07a8b](https://github.com/ulivz/vuepress-theme-blog/commit/7f07a8b))
* using GlobalLayout to simplify implementation ([8c89670](https://github.com/ulivz/vuepress-theme-blog/commit/8c89670))



# [1.0.0-beta.3](https://github.com/ulivz/vuepress-theme-blog/compare/v1.0.0-beta.2...v1.0.0-beta.3) (2019-06-04)


### Bug Fixes

* Cannot read property 'trim' of undefined ([0327584](https://github.com/ulivz/vuepress-theme-blog/commit/0327584))




example/.temp/internal/routes.js
/**
 * Generated by "@vuepress/internal-routes"
 */

import { injectComponentOption, ensureAsyncComponentsLoaded } from '@app/util'
import rootMixins from '@internal/root-mixins'
import GlobalLayout from "C:\\Git\\vuepress-theme-kb\\layouts\\GlobalLayout.vue"

injectComponentOption(GlobalLayout, 'mixins', rootMixins)
export const routes = [
  {
    name: "v-b07f4888",
    path: "/2018/11/07/frontmatter-in-vuepress-3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-b07f4888").then(next)
    },
  },
  {
    path: "/2018/11/07/frontmatter-in-vuepress-3/index.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress-3/"
  },
  {
    path: "/_posts/2018-11-7-frontmatter-in-vuepress-3.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress-3/"
  },
  {
    name: "v-a250cabc",
    path: "/2019/02/26/markdown-slot-2/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-a250cabc").then(next)
    },
  },
  {
    path: "/2019/02/26/markdown-slot-2/index.html",
    redirect: "/2019/02/26/markdown-slot-2/"
  },
  {
    path: "/_posts/2019-2-26-markdown-slot-2.html",
    redirect: "/2019/02/26/markdown-slot-2/"
  },
  {
    name: "v-7832e8be",
    path: "/2018/11/07/frontmatter-in-vuepress-2/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-7832e8be").then(next)
    },
  },
  {
    path: "/2018/11/07/frontmatter-in-vuepress-2/index.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress-2/"
  },
  {
    path: "/_posts/2018-11-7-frontmatter-in-vuepress-2.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress-2/"
  },
  {
    name: "v-4638da08",
    path: "/2018/11/07/frontmatter-in-vuepress/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-4638da08").then(next)
    },
  },
  {
    path: "/2018/11/07/frontmatter-in-vuepress/index.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress/"
  },
  {
    path: "/_posts/2018-11-7-frontmatter-in-vuepress.html",
    redirect: "/2018/11/07/frontmatter-in-vuepress/"
  },
  {
    name: "v-901caf40",
    path: "/2019/02/26/markdown-slot-3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-901caf40").then(next)
    },
  },
  {
    path: "/2019/02/26/markdown-slot-3/index.html",
    redirect: "/2019/02/26/markdown-slot-3/"
  },
  {
    path: "/_posts/2019-2-26-markdown-slot-3.html",
    redirect: "/2019/02/26/markdown-slot-3/"
  },
  {
    name: "v-7de893c4",
    path: "/2019/02/26/markdown-slot-4/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-7de893c4").then(next)
    },
  },
  {
    path: "/2019/02/26/markdown-slot-4/index.html",
    redirect: "/2019/02/26/markdown-slot-4/"
  },
  {
    path: "/_posts/2019-2-26-markdown-slot-4.html",
    redirect: "/2019/02/26/markdown-slot-4/"
  },
  {
    name: "v-c729b8a8",
    path: "/2019/02/26/markdown-slot/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-c729b8a8").then(next)
    },
  },
  {
    path: "/2019/02/26/markdown-slot/index.html",
    redirect: "/2019/02/26/markdown-slot/"
  },
  {
    path: "/_posts/2019-2-26-markdown-slot.html",
    redirect: "/2019/02/26/markdown-slot/"
  },
  {
    name: "v-2e581418",
    path: "/2019/05/06/writing-a-vuepress-theme-2/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-2e581418").then(next)
    },
  },
  {
    path: "/2019/05/06/writing-a-vuepress-theme-2/index.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-2/"
  },
  {
    path: "/_posts/2019-5-6-writing-a-vuepress-theme-2.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-2/"
  },
  {
    name: "v-186168f2",
    path: "/2019/05/06/writing-a-vuepress-theme-3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-186168f2").then(next)
    },
  },
  {
    path: "/2019/05/06/writing-a-vuepress-theme-3/index.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-3/"
  },
  {
    path: "/_posts/2019-5-6-writing-a-vuepress-theme-3.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-3/"
  },
  {
    name: "v-47eedbf0",
    path: "/2019/05/06/writing-a-vuepress-theme-4/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-47eedbf0").then(next)
    },
  },
  {
    path: "/2019/05/06/writing-a-vuepress-theme-4/index.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-4/"
  },
  {
    path: "/_posts/2019-5-6-writing-a-vuepress-theme-4.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme-4/"
  },
  {
    name: "v-69e22104",
    path: "/2019/05/06/writing-a-vuepress-theme/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Post", "v-69e22104").then(next)
    },
  },
  {
    path: "/2019/05/06/writing-a-vuepress-theme/index.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme/"
  },
  {
    path: "/_posts/2019-5-6-writing-a-vuepress-theme.html",
    redirect: "/2019/05/06/writing-a-vuepress-theme/"
  },
  {
    name: "v-47b1d5a7",
    path: "/about.html",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-47b1d5a7").then(next)
    },
  },
  {
    name: "v-b2754d56",
    path: "/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-b2754d56").then(next)
    },
    meta: {"pid":"post","id":"post"}
  },
  {
    path: "/index.html",
    redirect: "/"
  },
  {
    name: "v-b1564aac",
    path: "/tag/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterKey", "v-b1564aac").then(next)
    },
    meta: {"pid":"tag","id":"tag"}
  },
  {
    path: "/tag/index.html",
    redirect: "/tag/"
  },
  {
    name: "v-f8241222",
    path: "/tag/frontmatter/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterPagination", "v-f8241222").then(next)
    },
    meta: {"pid":"tag","id":"frontmatter"}
  },
  {
    path: "/tag/frontmatter/index.html",
    redirect: "/tag/frontmatter/"
  },
  {
    name: "v-dfb71430",
    path: "/tag/vuepress/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterPagination", "v-dfb71430").then(next)
    },
    meta: {"pid":"tag","id":"vuepress"}
  },
  {
    path: "/tag/vuepress/index.html",
    redirect: "/tag/vuepress/"
  },
  {
    name: "v-3ae5b494",
    path: "/tag/markdown/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterPagination", "v-3ae5b494").then(next)
    },
    meta: {"pid":"tag","id":"markdown"}
  },
  {
    path: "/tag/markdown/index.html",
    redirect: "/tag/markdown/"
  },
  {
    name: "v-58f012a0",
    path: "/tag/theme/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterPagination", "v-58f012a0").then(next)
    },
    meta: {"pid":"tag","id":"theme"}
  },
  {
    path: "/tag/theme/index.html",
    redirect: "/tag/theme/"
  },
  {
    name: "v-1559f73a",
    path: "/tag/blog/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("FrontmatterPagination", "v-1559f73a").then(next)
    },
    meta: {"pid":"tag","id":"blog"}
  },
  {
    path: "/tag/blog/index.html",
    redirect: "/tag/blog/"
  },
  {
    name: "v-8fdafa70",
    path: "/page/2/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-8fdafa70").then(next)
    },
    meta: {"pid":"post","id":"post"}
  },
  {
    path: "/page/2/index.html",
    redirect: "/page/2/"
  },
  {
    name: "v-8fdafa32",
    path: "/page/3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-8fdafa32").then(next)
    },
    meta: {"pid":"post","id":"post"}
  },
  {
    path: "/page/3/index.html",
    redirect: "/page/3/"
  },
  {
    name: "v-c5eeaa56",
    path: "/tag/vuepress/page/2/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-c5eeaa56").then(next)
    },
    meta: {"pid":"tag","id":"vuepress"}
  },
  {
    path: "/tag/vuepress/page/2/index.html",
    redirect: "/tag/vuepress/page/2/"
  },
  {
    name: "v-c5eeaa18",
    path: "/tag/vuepress/page/3/",
    component: GlobalLayout,
    beforeEnter: (to, from, next) => {
      ensureAsyncComponentsLoaded("Layout", "v-c5eeaa18").then(next)
    },
    meta: {"pid":"tag","id":"vuepress"}
  },
  {
    path: "/tag/vuepress/page/3/index.html",
    redirect: "/tag/vuepress/page/3/"
  },
  {
    path: '*',
    component: GlobalLayout
  }
]